tasks/review-app.js

```diff
@@ -0,0 +1,173 @@
+	}
+	return res;
+};
+
+const getGithubArchiveRedirectUrl = ({ repoName, branch, githubToken }) => {
+	const url = getGithubArchiveUrl({ repoName, branch });
+
+	return fetch(url, {
+		headers: {
+			Authorization: `token ${githubToken}`
+		},
+		redirect: 'manual' // Don't follow redirect, just want the URL
+	}).then(res => {
+		if (res.status !== 302) {
+			throw new Error(`Unexpected response for ${url} (${status})`);
+		}
+
+		const { headers: { _headers: { location } } } = res;
+		const [ redirectUrl ] = location || [];
+
+		return redirectUrl;
+	});
+};
+
+const waitTillReviewAppCreated = (data) => {
+	const { id } = data;
+	const checkForCreatedStatus = async () => {
+		const headers = await herokuHeaders({ useReviewAppApi: true });
+		const result = await fetch(getReviewAppUrl(id), {
+			headers
+		})
+			.then(throwIfNon2xx)
+			.then(res => res.json())
+			.then(data => {
+				const { status, message, app } = data;
+
+				if (status === REVIEW_APP_STATUSES.deleted) {
+					throw new pRetry.AbortError(`Review app was deleted: ${message}`);
+				}
+
+				if (status !== REVIEW_APP_STATUSES.created) {
+					const appIdOutput = (status === REVIEW_APP_STATUSES.creating)
+						? `, appId: ${app.id}`
+						: '';
+					throw new Error(`Review app not created yet. Current status: ${status}${appIdOutput}`);
+				};
+
+				return app.id;
+			});
+		return result;
+	};
+
+	return pRetry(checkForCreatedStatus, {
+		factor: RETRY_EXP_BACK_OFF_FACTOR,
+		retries: NUM_RETRIES,
+		minTimeout: RETRY_INTERVAL,
+		onFailedAttempt: (err) => {
+			const { attemptNumber, message } = err;
+			console.error(`${attemptNumber}/${NUM_RETRIES}: ${message}`); // eslint-disable-line no-console
+		}
+	});
+};
+
+const getAppName = async (appId) => {
+	const headers = await herokuHeaders();
+	return fetch(getAppUrl(appId), {
+		headers
+	})
+		.then(throwIfNon2xx)
+		.then(res => res.json())
+		.then((result) => {
+			const { name } = result;
+			return name;
+		});
+};
+
+async function task (app, options) {
+	const { repoName, branch, commit, githubToken } = options;
+
+	const { id: pipelineId } = await pipelineInfo(app);
+	const headers = await herokuHeaders({ useReviewAppApi: true });
+	const body = {
+		pipeline: pipelineId,
+		branch,
+		source_blob: {
+			url: await getGithubArchiveRedirectUrl({ repoName, branch, githubToken }),
+			version: commit
+		}
+	};
+
+	return fetch(REVIEW_APPS_URL, {
+		headers,
+		method: 'post',
+		body: JSON.stringify(body)
+	})
+		.then(throwIfNon2xx)
+		.then(res => res.json())
+		.then(waitTillReviewAppCreated)
+		.then(getAppName)
+		.then(appName => {
+			console.log(appName); // eslint-disable-line no-console
+		});
+}
+
+/**
+* Assume
+* 	* app is VAULT_SOURCE, and is package.json name (could assume it's the package.json name, like `nht configure`)
+*/
+module.exports = function (program) {
+	program
+		.command('review-app [app]')
+		.description('Create a heroku review app and print out the app name created')
```

Comment:
Perhaps this should say something like "Create or retrieve a heroku review app for a given branch, and print out the review app name" - to make it explicit that it won't recreate.

Reply:
I think the redeploy was only for the normal heroku review app, for the review apps beta, you need to explicitly delete before you can build again. I just tried it with different commits (even if it is a subsequent commit in the branch)

I'm gonna add a bit to delete it first if it already exists.

Reply:
Hmm. Is that true even if there is a PR open? I can _swear_ that it seemed to rebuild when we tried it 🤔

Reply:
Yep, there's a PR here: https://github.com/Financial-Times/next-search-page/pull/158

But it might be something up with my test environment. To test properly, I'm going to need to run it on CI. TBC.

Reply:
Fixed it so that if there is an existing review app, it deletes it first and prints out `Review app already created for post-build branch. Deleting existing review app first.`, then creates a new review app.

You can see it running here: https://circleci.com/gh/Financial-Times/next-search-page/1634